jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/StacklessLogging.java

```diff
@@ -0,0 +1,126 @@
+ * <p>
+ * Use of other logging implementation cause no effect when using this class
+ * <p>
+ * Example:
+ *
+ * <pre>
+ * try (StacklessLogging scope = new StacklessLogging(EventDriver.class,Noisy.class))
+ * {
+ *     doActionThatCausesStackTraces();
+ * }
+ * </pre>
+ */
+public class StacklessLogging implements AutoCloseable
+{
+    private static final Logger LOG = LoggerFactory.getLogger(StacklessLogging.class);
+    private static final JettyLoggerFactory loggerFactory;
+
+    static
+    {
+        JettyLoggerFactory jettyLoggerFactory = null;
+        ILoggerFactory activeLoggerFactory = LoggerFactory.getILoggerFactory();
+        if (activeLoggerFactory instanceof JettyLoggerFactory)
+        {
+            jettyLoggerFactory = (JettyLoggerFactory)activeLoggerFactory;
+        }
+        else
+        {
+            LOG.warn("Unable to squelch stacktraces ({} is not a {})",
+                activeLoggerFactory.getClass().getName(),
+                JettyLoggerFactory.class.getName());
+        }
+        loggerFactory = jettyLoggerFactory;
+    }
+
+    private final Set<JettyLogger> squelched = new HashSet<>();
+
+    public StacklessLogging(Class<?>... classesToSquelch)
+    {
+        for (Class<?> clazz : classesToSquelch)
+        {
+            JettyLogger jettyLogger = loggerFactory.getJettyLogger(clazz.getName());
+            // only operate on loggers that are of type StdErrLog
+            if (!jettyLogger.isDebugEnabled())
+            {
+                if (!jettyLogger.isHideStacks())
+                {
+                    jettyLogger.setHideStacks(true);
+                    squelched.add(jettyLogger);
+                }
+            }
+        }
+    }
+
+    public StacklessLogging(Package... packagesToSquelch)
+    {
+        for (Package pkg : packagesToSquelch)
+        {
+            JettyLogger jettyLogger = loggerFactory.getJettyLogger(pkg.getName());
+            // only operate on loggers that are of type StdErrLog
+            if (!jettyLogger.isDebugEnabled())
+            {
+                if (!jettyLogger.isHideStacks())
+                {
+                    jettyLogger.setHideStacks(true);
+                    squelched.add(jettyLogger);
+                }
+            }
+        }
+    }
+
+    public StacklessLogging(Logger... logs)
+    {
+        for (Logger log : logs)
+        {
+            // only operate on loggers that are of type StdErrLog
+            if (log instanceof JettyLogger && !log.isDebugEnabled())
+            {
+                JettyLogger jettyLogger = ((JettyLogger)log);
+                if (!jettyLogger.isHideStacks())
+                {
+                    jettyLogger.setHideStacks(true);
+                    squelched.add(jettyLogger);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void close()
+    {
+        for (JettyLogger log : squelched)
+        {
+            log.setHideStacks(false);
```

Comment:
We do not remember if the stack were hidden before? We should restore to whatever value it was before, rather than forcing `false`, no? Make `squelched` a `Map` and we should be good.

Reply:
Good point.
But to be fair, we only really use StacklessLogging in our unit tests.

Reply:
No work is needed here.
The `squelched` collection only has loggers that were originally `logger.isHideStacks == false;`
Entries that were originally `logger.isHideStacks == true` are not present in the `squelched` list.